geode-core/src/main/java/org/apache/geode/management/internal/api/LocatorClusterManagementService.java

```diff
@@ -202,11 +214,19 @@ public LocatorClusterManagementService(InternalConfigurationPersistenceService p
     }
 
     // persist configuration in cache config
+    List<String> updatedGroups = new ArrayList<>();
+    List<String> failedGroups = new ArrayList<>();
     for (String groupName : groups) {
-      configurationManager.add(config, groupName);
+      try {
+        configurationManager.add(config, groupName);
+        updatedGroups.add(groupName);
+      } catch (Exception e) {
```

Comment:
Could we make some of the new "catch (Exception ...)" more specific? Do we not know what exceptions we expect configurationManager.add to throw?

Reply:
it depends on each configuration manager's implementation. I don't think we can dictate on what kind of exception it can throw. I added this signature here so that the handling of the exception falls on the caller, not the implementer.

Reply:
If this is an SPI then this interface should dictate what the implementor needs to do. It already says you have to implement an "add" method etc. It s reasonable if we want to support exceptions from some of these methods to say so on the interface and then the implementer has to make sure and throw that type of exception. That would let the implementor explicitly say "I could not do an add for the following reason" and we could treat that as a handled exception. Other exceptions (like say an NPE caused by their coding error) could then be highlighted as unexpected. But with this code we have to treat all of them as unexpected and they end up with a big call stack in the log which is not needed if it is just something like "I did not have file system permissions".